src/lib/setup/servicePrincipalService.ts

```diff
@@ -0,0 +1,33 @@
+import { logger } from "../../logger";
+import { exec } from "../shell";
+import { IRequestContext } from "./constants";
+
+export const azCLILogin = async () => {
+  try {
+    logger.info("attempting to login to az command line");
+    await exec("az", ["login"]);
+    logger.info("Successfully login to az command line");
+  } catch (err) {
+    logger.error("Unable to execute az login");
+    logger.error(err);
+    throw err;
+  }
+};
+
+export const createWithAzCLI = async (rc: IRequestContext) => {
+  await azCLILogin();
+  try {
+    logger.info("attempting to create service principal with az command line");
+    const result = await exec("az", ["ad", "sp", "create-for-rbac"]);
```

Comment:
Is there a SDK available to manage SP instead of using az cli?

As an example: https://docs.microsoft.com/en-us/azure/javascript/node-sdk-azure-authenticate-principal#create-a-service-principal-using-the-azure-sdk-for-nodejs

Reply:
in order to user the SDK, we need to authenticate and hence we need a service principal. So we need to create a service principal or ask user to enter service principal id, password and tenantId. Once this is done, we can start to use the SDK.